dali/kernels/common/join/tensor_join_cpu.h

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_TENSOR_JOIN_CPU_H
+#define DALI_TENSOR_JOIN_CPU_H
+
+#include "dali/kernels/kernel.h"
+#include "dali/core/tensor_shape_print.h"
+#include "dali/core/format.h"
+
+namespace dali {
+namespace kernels {
+
+enum JoinMode {
+  STACK,
+  CONCAT
+};
+
+namespace detail {
+
+template<typename T>
+void TransferBuffers(span<T> output, const TensorShape<> &output_shape,
+                     span<TensorView<StorageCPU, const T>> inputs, int axis) {
+  vector<int64_t> copy_sizes(inputs.size());
```

Comment:
Perhaps a SmallVector? The number of inputs will be fixed at pipeline build an it's likely to be quite small. I think that `SmallVector<int64_t, 8>` should suffice.